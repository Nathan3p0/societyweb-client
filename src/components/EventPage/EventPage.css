.eventPage__main {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    padding: 2rem;
}

.eventPage__event-attending {
    background-color: #ddd;
    padding: 1rem;
}

.eventPage__event-info {
    padding: 1rem;
    background-color: #333;
}

.eventPage__event-info h3 {
    font-size: 2rem;
}

.eventPage__event-info button {
    margin-top: 2rem;
}