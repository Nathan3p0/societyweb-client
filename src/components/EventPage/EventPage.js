import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import moment from 'moment';
import AttendingMemberList from '../AttendingMembersList/AttendingMembersList';
import AdminApiService from '../../services/admin-api-service';
import './EventPage.css';

const EventPage = (props) => {
    let history = useHistory();
    const { id } = props.match.params;

    const [error, setError] = useState(null);
    const [event, setEvent] = useState({ event_name: 'loading' });

    const fetchEvents = () => {
        AdminApiService.fetchEventById(id)
            .then(res =>
                setEvent(res)
            )
            .catch(err =>
                setError(err)
            )
    }

    useEffect(() => {
        fetchEvents()
    }, [])

    const handleGoBack = () => {
        history.goBack();
    }
    const date = new Date(event.event_date)
    return (
        <>
            <section className="eventPage__main">
                <section className="eventPage__event-info">
                    {error && <p className="error">{error}</p>}
                    <h3>Event: {event.event_name}</h3>
                    <div className="eventPage__event-info--content">
                        <p><strong>Date:</strong> {moment(date).format('MMMM Do YYYY')}</p>
                        <p><strong>Time:</strong> {event.event_time}</p>
                        <p><strong>Location:</strong> {event.event_location}</p>
                        <p><strong>Description:</strong> {event.event_description}</p>
                        <button onClick={handleGoBack}>Go Back</button>
                    </div>
                </section>
                <section className="eventPage__event-attending">
                    <AttendingMemberList id={id} />
                </section>
            </section>
        </>
    );
}

export default EventPage;