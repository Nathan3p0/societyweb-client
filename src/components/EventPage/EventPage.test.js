import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom';
import EventPage from './EventPage'

describe('Event Page Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<MemoryRouter><EventPage match={{ params: { id: 1 } }} /></MemoryRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<MemoryRouter><EventPage match={{ params: { id: 1 } }} /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})