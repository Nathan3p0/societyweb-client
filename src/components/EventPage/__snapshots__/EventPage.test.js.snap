// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Page Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="eventPage__main"
>
  <section
    className="eventPage__event-info"
  >
    <h3>
      Event: 
      loading
    </h3>
    <div
      className="eventPage__event-info--content"
    >
      <p>
        <strong>
          Date:
        </strong>
         
        November 21st 2019
      </p>
      <p>
        <strong>
          Time:
        </strong>
         
      </p>
      <p>
        <strong>
          Location:
        </strong>
         
      </p>
      <p>
        <strong>
          Description:
        </strong>
         
      </p>
      <button
        onClick={[Function]}
      >
        Go Back
      </button>
    </div>
  </section>
  <section
    className="eventPage__event-attending"
  >
    <section>
      <h3>
        Members Attending Event:
      </h3>
      <ul
        className="responsiveTable"
      >
        <li
          className="tableHeader"
        >
          <div
            className="col col-1"
          >
            Name
          </div>
          <div
            className="col col-2"
          >
            Phone
          </div>
          <div
            className="col col-3"
          >
            Email
          </div>
          <div
            className="col col-4"
          >
            RSVP Status
          </div>
        </li>
      </ul>
    </section>
  </section>
</section>
`;
